policy/resolver.go

```diff
@@ -1578,6 +1588,41 @@ func (s *LocalServices) jobsToControls(cache *frameworkResolverCache, framework
 				rj.Notify = append(rj.Notify, queryJob.Uuid)
 				continue
 			}
+
+		case ResolvedFrameworkNodeTypeQuery:
+			if len(targets) == 0 {
+				continue
+			}
+
+			// the query must exist, since validation happens earlier
+			mquery := cache.bundleMap.Queries[mrn]
+			execQuery := cache.executionQueries[mquery.Checksum]
+			uuid := cache.relativeChecksum(mquery.Mrn)
+			queryJob := &ReportingJob{
+				Uuid:      uuid,
+				QrId:      mquery.Mrn,
+				ChildJobs: map[string]*explorer.Impact{},
+				Type:      ReportingJob_DATA_QUERY,
+			}
+			nuJobs[uuid] = queryJob
+			for controlMrn := range targets {
+				controlJob := ensureControlJob(cache, nuJobs, controlMrn, framework, frameworkGroupByControlMrn)
+				controlJob.ChildJobs[queryJob.Uuid] = nil
+				queryJob.Notify = append(queryJob.Notify, controlJob.Uuid)
+				err := connectDatapointsToReportingJob(execQuery, controlJob, job.Datapoints)
+				if err != nil {
+					return err
+				}
+				// feed the datapoints into the query's reporting job too, not just the control job
+				err = connectDatapointsToReportingJob(execQuery, queryJob, job.Datapoints)
```

Comment:
I think we'd only need to add them to the queryJob, right? Because the queryjob feeds into the control job, so the control job gets its data from that if I'm not mistaken

Reply:
Good point, I moved this to only report to the query job: datapoint -> query job -> control job